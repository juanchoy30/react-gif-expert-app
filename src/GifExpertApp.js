import React, { useState } from 'react'
import AddCategory from './components/AddCategory';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    /*
    const handleAdd = () => {
        const Addition = 'Banjo & Kazooie'

        setCategories( cats => [ ...cats, Addition] );  // cats is the previous value

    };
    */

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr />

            <ol>
                {
                    categories.map( category => {
                        return <li key={ category }> { category } </li> // The key is what tells react the position of the index
                    })
                }
            </ol>
        </>
    );
}

export default GifExpertApp