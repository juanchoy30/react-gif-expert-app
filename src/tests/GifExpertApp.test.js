import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('<GifExpertApp /> Tests', () => {

    test('should render the component correctly', () => {
            
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('should show a list of categories', () => {
        
        const categories = ['Mr Robot', 'X Files'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });

})
