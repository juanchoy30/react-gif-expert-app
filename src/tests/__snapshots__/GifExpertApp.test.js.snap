// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> Tests should render the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp /> Tests should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Mr Robot"
      key="Mr Robot"
    />
    <GifGrid
      category="X Files"
      key="X Files"
    />
  </ol>
</Fragment>
`;
